text/0000-std-aware-cargo.md

```diff
@@ -0,0 +1,335 @@
+
+This functionality would be possible even with the use of a stable compiler.
+
+Users of a nightly compiler would be able to set compile time feature flags for `core` through settings made in their `Cargo.toml`.
+
+### Caveats
+
+For users of a stable compiler, it would not be possible to modify the source code contents of `core`, or change any compile time features of `core` from the defaults used when publishing pre-compiled versions of `core`.
+
+The source code used to build `core` would be the same as the compiler used for building the current project.
+
+### User Interaction
+
+When compiling for a non-standard target, users may specify their target using a target specification file, rather than a pre-defined target.
+
+> NOTE: The current target specification is described in JSON, and contains some
+> implementation details regarding the use of LLVM as the compiler backend. This
+> RFC does not prescribe any changes to the Target Specification format, and is
+> intended to work with whatever the current/stable method of specifying a
+> custom target is.
+
+For example, currently a user may cross-compile by specifying a target known by Rust:
+
+```sh
+cargo build --target thumbv7em-none-eabihf
+```
+
+Users would also be able to specify a target specification file, by providing a path to the file to be used.
+
+```sh
+cargo build --target thumbv7em-freertos-eabihf.json
+```
+
+In general, any of the following would prompt Cargo to recompile `core`, rather than use a pre-compiled version:
+
+* A custom target specification is used
+* The root crate has modified the feature flags of `core`
+* The root crate has set certain profile settings, such as opt-level, etc.
+* The root crate has specified a `patch.sysroot` (this is defined in a later section)
+
+Users of a stable compiler would not be able to customize `core` outside of these profile settings.
+
+For users of a nightly compiler, compile time features of `core` may be specified using the same syntax used for other crate dependencies. These specified features may include unstable features.
+
+```toml
+[dependencies.core]
+default-features = false
+features = [...]
+```
+
+It is not necessary to explicitly mention the dependency of `core`, unless changes to features are necessary.
+
+Cargo would use the source of `core` located in the user's `SYSROOT` directory. This source code would be obtained in the same way as necessary today, through the use of `rustup component add rust-src`. If this component is missing, Cargo would exit with an error code, and would prompt the user to execute the command specified above.
+
+### Technical Implications
+
+#### Stabilization of a Target Specification Format
+
+As the custom target specifications (currently JSON) would become part of the stable interface of Cargo. The format used by this file must become stabilized, and further changes must be made in a backwards compatible way to guarantee stability.
```

Comment:
Nobody contradicted me when I said this can be a black-box the Cargo compares byte-by-byte as a stop-gap. I think trading some cache hits for future compat is a good move here.